aevnmt/models/aevnmt.py

```diff
@@ -78,7 +90,10 @@ def inference_parameters(self):
     def generative_parameters(self):
         # TODO: separate the generative model into a GenerativeModel module
         #  within that module, have two modules, namely, LanguageModel and TranslationModel
-        return chain(self.lm_parameters(), self.tm_parameters())
+        return chain(self.lm_parameters(), self.tm_parameters(), self.bow_parameters())
+
+    def bow_parameters(self):
+        return chain( iter(()) if self.bow_output_layer is None else self.bow_output_layer.parameters()   , iter(()) if self.bow_output_layer_tl is None else self.bow_output_layer_tl.parameters()  )
```

Comment:
Please write this in regular if-statements as this is quite hard to follow.

Reply:
Done, added new commit with regular ifs.